packages/react/src/components/Tree/Tree.tsx

```diff
@@ -0,0 +1,331 @@
+
+import TreeItem, { TreeItemProps } from './TreeItem'
+import {
+  childrenExist,
+  commonPropTypes,
+  createShorthandFactory,
+  UIComponentProps,
+  ChildrenComponentProps,
+  rtlTextContainer,
+  applyAccessibilityKeyHandlers,
+  AutoControlledComponent,
+} from '../../lib'
+import {
+  ShorthandRenderFunction,
+  WithAsProp,
+  withSafeTypeForAs,
+  ShorthandCollection,
+  ShorthandValue,
+} from '../../types'
+import { Accessibility } from '../../lib/accessibility/types'
+import { treeBehavior } from '../../lib/accessibility'
+import { getNextElement } from '../../lib/accessibility/FocusZone/focusUtilities'
+
+export interface TreeSlotClassNames {
+  item: string
+}
+
+export interface TreeProps extends UIComponentProps, ChildrenComponentProps {
+  /** Accessibility behavior if overridden by the user. */
+  accessibility?: Accessibility
+
+  /** Map with the subtrees and information related to their open/closed state. */
+  activeItemIds?: string[]
+
+  /** Initial activeIndex value. */
+  defaultActiveItemIds?: string[]
+
+  /** Only allow one subtree to be open at a time. */
+  exclusive?: boolean
+
+  /** Shorthand array of props for Tree. */
+  items?: ShorthandCollection<TreeItemProps>
+
+  /**
+   * A custom render function for the title slot.
+   *
+   * @param {React.ReactType} Component - The computed component for this slot.
+   * @param {object} props - The computed props for this slot.
+   * @param {ReactNode|ReactNodeArray} children - The computed children for this slot.
+   */
+  renderItemTitle?: ShorthandRenderFunction
+}
+
+export interface TreeItemForRenderProps {
+  elementRef: React.RefObject<HTMLElement>
+  id: string
+  index: number
+  level: number
+  parent: ShorthandValue<TreeItemProps>
+  siblings: ShorthandCollection<TreeItemProps>
+}
+
+export interface TreeState {
+  activeItemIds: string[]
+  itemsForRender: Record<string, TreeItemForRenderProps>
+}
+
+class Tree extends AutoControlledComponent<WithAsProp<TreeProps>, TreeState> {
+  static create: Function
+
+  static displayName = 'Tree'
+
+  static className = 'ui-tree'
+
+  static slotClassNames: TreeSlotClassNames = {
+    item: `${Tree.className}__item`,
+  }
+
+  static propTypes = {
+    ...commonPropTypes.createCommon({
+      content: false,
+    }),
+    activeItemIds: customPropTypes.collectionShorthand,
+    defaultActiveItemIds: customPropTypes.collectionShorthand,
+    exclusive: PropTypes.bool,
+    items: customPropTypes.collectionShorthand,
+    renderItemTitle: PropTypes.func,
+    rtlAttributes: PropTypes.func,
+  }
+
+  static defaultProps = {
+    as: 'div',
+    accessibility: treeBehavior,
+  }
+
+  static autoControlledProps = ['activeItemIds']
+
+  static hasSubtree(item: TreeItemProps | ShorthandValue<TreeItemProps>): boolean {
+    return !_.isNil(item['items']) && item['items'].length > 0
+  }
+
+  static getItemsForRender = _.memoize((itemsFromProps: ShorthandCollection<TreeItemProps>) => {
```

Comment:
```suggestion
  static getItemsForRender = (itemsFromProps: ShorthandCollection<TreeItemProps>) => {
```

I am really not sure that `_.memoize()` will work properly with nested objects. I suggest to remove it until we will meet a performance bottleneck